src/algebra/big_operators/finprod.lean

```diff
@@ -876,13 +876,20 @@ iterating this lemma, e.g., if we have `f : α × β × γ → M`. -/
 begin
   have : ∀ a, ∏ (i : β) in (s.filter (λ ab, prod.fst ab = a)).image prod.snd, f (a, i) =
     (finset.filter (λ ab, prod.fst ab = a) s).prod f,
-  { intros a, apply finset.prod_bij (λ b _, (a, b)); finish, },
+  { intros a, apply finset.prod_bij (λ b _, (a, b)), -- `finish` could close the goals here
+  { suffices : ∀ (a_1 : α) (b : β), (a_1, b) ∈ s → a_1 = a → (a, b) ∈ s ∧ a_1 = a, { simp },
+    intros a' b p ha, rw ←ha, use p },
+  { simp },
+  { simp },
+  { suffices : ∀ (a_1 : α) (b : β), (a_1, b) ∈ s → a_1 = a → (a, b) ∈ s ∧ a_1 = a, { simpa },
+    intros a' b p ha, rw ←ha, use p } },
```

Comment:
```suggestion
  { refine λ a, finset.prod_bij (λ b _, (a, b)) (by simp) (by simp) (by simp) _,
    -- `finish` could close the goal here
    suffices : ∀ (a' : α) (b : β), (a', b) ∈ s → a' = a → (a, b) ∈ s ∧ a' = a, by simpa,
    { rintro a' b p rfl,
      exact ⟨p, rfl⟩ } },
```
Minor golfing.

Reply:
Thanks to your comment I've been able to golf this a bit further.